core/metacling/src/TCling.cxx

```diff
@@ -555,15 +555,16 @@ void TCling::HandleNewDecl(const void* DV, bool isDeserialized, std::set<TClass*
       // scope.
       if (!(isa<VarDecl>(ND)))
          return;
-
+      auto ListOfGlobals = static_cast<TListOfDataMembers*>(gROOT->GetListOfGlobals());
+      const VarDecl *CanonVD = cast<VarDecl>(ND)->getCanonicalDecl();
       // Skip if already in the list.
-      if (gROOT->GetListOfGlobals()->FindObject(ND->getNameAsString().c_str()))
+      if (ListOfGlobals->Find(static_cast<DeclId_t>(CanonVD)))
```

Comment:
Would be it be better as a const_cast and/or an update to TListOfDataMembers::Find to take a const decl?

Reply:
I am not sure I understand the `const_cast` comment. I agree we may want to update `TListOfDataMembers::Find` as a next step in improving this code.

Reply:
My bad. I misread the line number in the error messages.